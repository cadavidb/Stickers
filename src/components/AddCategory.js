import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <>
        <div className="background"></div>
        <div id="entrada">
        <form onSubmit={ handleSubmit }>
            <p> { inputValue } </p>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
      
        </div>
        
        <button onClick={handleSubmit} id="boton" className="animate__animated animate__shakeY">Buscar Gifs</button>
        </>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
