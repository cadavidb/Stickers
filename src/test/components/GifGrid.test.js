import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas sobre componente GifGrid', () => {

    const categoria="ninjas"
    //falseamos la data 
    useFetchGifs.mockReturnValue({
        data:[],
        loading:true
    })

//copiamos el componente
    const wrapper=shallow(<GifGrid category={categoria}/>)

//testeamos que este igual
    test('debe hacer un match con el snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    
    test('debe de mostrar item cuando se cargan imagenes useFecthGifs', () => {
        const data=[{
            id:'ACB',
            url:'http',
            title:'anything'
        }]


        useFetchGifs.mockReturnValue({
            data:data,
            loading:false
        })
        const wrapper=shallow(<GifGrid category={categoria}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(data.length)

    })
    
})
