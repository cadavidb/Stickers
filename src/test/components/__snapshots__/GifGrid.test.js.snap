// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre componente GifGrid debe de mostrar item cuando se cargan imagenes useFecthGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    ninjas
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ACB"
      key="ACB"
      title="anything"
      url="http"
    />
  </div>
</Fragment>
`;

exports[`Pruebas sobre componente GifGrid debe hacer un match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    ninjas
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
